packages/material-ui/src/Modal/Modal.js

```diff
@@ -232,6 +219,7 @@ class Modal extends React.Component {
 
     // It's a Transition like component
     if (hasTransition) {
+      childProps.onEnter = createChainedFunction(this.handleEnter, children.props.onEnter);
```

Comment:
What do you think of using `enter` instead of `entered`? It seems to do the job better :).

Reply:
I think the same applies here https://github.com/mui-org/material-ui/pull/15250#discussion_r273253921 - Should stop it from abruptly unmounting during the appear / enter animation.
Does it help with any of the failed tests, too?

Reply:
Yes, I think that we can apply the same change to the Popper component :).